sc/social/like/tests/test_canonicalurl_view.py

```diff
@@ -72,3 +73,33 @@ def test_update_canonical_url(self):
         # canonical URL unchaged
         self.assertEqual(
             self.portal['baz'].canonical_url, 'https://example.org/plone/baz')
+
+
+class UtilPathToVirtualPathTestCase(unittest.TestCase):
+
+    layer = INTEGRATION_TESTING
+
+    def setUp(self):
+        app = self.layer['app']
+        portal = self.layer['portal']
+        self.request = self.layer['request']
+
+        if 'virtual_hosting' not in app.objectIds():
+            manage_addVirtualHostMonster(app, 'virtual_hosting')
+
+        with api.env.adopt_roles(['Manager']):
+            self.obj = api.content.create(portal, type='News Item', id='foo')
+            api.content.transition(self.obj, 'publish')
+
+    def test_get_path_to_virtual_path(self):
+        self.request.traverse('/plone/')
+        self.assertEqual(
+            get_path_to_virtual_path(self.request, self.obj),
+            'plone/foo')
+
+    def test_get_path_to_virtual_path_using_virtualhostmonster(self):
```

Comment:
this test must be moded the `test_behaviors.py` module; rename it to `test_canonical_url_with_virtual_host_root`.